server/src/test/java/com/google/sps/AdRowProcessorTest.java

```diff
@@ -1,30 +1,186 @@
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void Should_ThrowException_When_IncorrectNumberOfArguments() {
+    String[] row = {"Id","Advertiser"};
+    Ad ad = AdRowProcessor.convertRowToAd(row);
+  }
+
+  @Test(expected = DateTimeParseException.class)
+  public void Should_ThrowException_When_StartDateNotIsostringFormat() {
+    String badStartDate = "2018-22-06";
+    String[] row = {"Id","Advertiser",badStartDate,"2000-01-01","≤ 10k",
+            "Not targeted","Not targeted","Not targeted","0","0",
+            "Headline", "Ad Link","Content", "Headline Sentiment", "Headline Terms", 
+            "Content Sentiment", "Content Terms"};
+    Ad ad = AdRowProcessor.convertRowToAd(row);
+  }
+
+  @Test(expected = DateTimeParseException.class)
+  public void Should_ThrowException_When_EndDateNotIsostringFormat() {
+    String badEndDate = "2018.11.06";
+    String[] row = {"Id","Advertiser","2000-01-01",badEndDate,"≤ 10k",
+            "Not targeted","Not targeted","Not targeted","0","0",
+            "Headline", "Ad Link","Content", "Headline Sentiment", "Headline Terms", 
+            "Content Sentiment", "Content Terms"};
+    Ad ad = AdRowProcessor.convertRowToAd(row);
+  }
+
+  @Test
+  public void Should_ProcessAgeTargetWithoutError_When_TargetHasWhitespace() {
+    String untargeted = "    Not Targeted  ";
+    String[] row1 = {"Id","Advertiser","2000-01-01","2000-01-01","≤ 10k",untargeted,
+                    "Not targeted","Not targeted","0","0","Headline", "Ad Link",
+                    "Content", "Headline Sentiment", "Headline Terms", "Content Sentiment",
+                    "Content Terms"}; 
+    Ad ad1 = AdRowProcessor.convertRowToAd(row1);
+    Assert.assertEquals(ad1.getIsTargetingAge(), false); 
+
+    String targeted = " 18-48 ";
+    String[] row2 = {"Id","Advertiser","2000-01-01","2000-01-01","≤ 10k",targeted,
+                    "Not targeted","Not targeted","0","0","Headline", "Ad Link",
+                    "Content", "Headline Sentiment", "Headline Terms", "Content Sentiment",
+                    "Content Terms"}; 
+    Ad ad2 = AdRowProcessor.convertRowToAd(row2);
+    Assert.assertEquals(ad2.getIsTargetingAge(), true);   
+  }
+
+  @Test
+  public void Should_AcceptTarget_When_GenderTargetValid() {
+    String validInput = " Female, Unknown gender ";
+    List<String> expected = Arrays.asList("Female","Unknown gender");
+    String[] row = {"Id","Advertiser","2000-01-01","2000-01-01","≤ 10k","Not targeted",
+                    validInput,"Not targeted","0","0","Headline", "Ad Link",
+                    "Content", "Headline Sentiment", "Headline Terms", "Content Sentiment",
+                    "Content Terms"}; 
+    Ad ad = AdRowProcessor.convertRowToAd(row);
+    Assert.assertEquals(ad.getGenderTargets(), expected); 
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void Should_ThrowException_When_GenderTargetInvalid() {
+    String invalidInput = " apples ";
```

Comment:
Write your test in 3 stanzas.

1: Setup
2: Execution
3: Test expected result

With new line between every stanza.